File: k6/pre.js
const http = require("http");

const options = {
  hostname: 'localhost',
  port: 8080,
  path: '/api/v1/ad',
  method: 'POST',
  headers: {
    'Content-Type': 'application/json',
  },
};

// 產生 [min, max] 的隨機整數
function randomInt(min, max) {
  return Math.floor(Math.random() * (max - min + 1) + min);
}

const genderSets = [[], ["M"], ["F"]];
const countrySets = [[], ["TW"], ["JP"], ["TW", "JP"]];
const platformSets = [
  [],
  ["android"],
  ["ios"],
  ["web"],
  ["android", "ios"],
  ["ios", "web"],
  ["web", "android"],
  ["android", "ios", "web"],
];

for (let i = 0; i < 1000; i++) {
  const ageStart = randomInt(1, 100);
  const now = new Date();
  const todayStart = new Date(now.getFullYear(), now.getMonth(), now.getDate());
  const todayEnd = new Date(
    now.getFullYear(),
    now.getMonth(),
    now.getDate(),
    23,
    59,
    59,
    999
  );

  let postData = {
    title: `AD ${i}`,
    startAt: todayStart.toISOString(),
    endAt: todayEnd.toISOString(),
    conditions: [
      {
        ageStart: ageStart,
        ageEnd: randomInt(ageStart, 100),
        gender: genderSets[randomInt(0, genderSets.length - 1)],
        country: countrySets[randomInt(0, countrySets.length - 1)],
        platform: platformSets[randomInt(0, platformSets.length - 1)],
      },
    ],
  };

  const req = http.request(options, (res) => {
    let data = '';
    res.on('data', (chunk) => {
      data += chunk;
    });
    res.on('end', () => {
      console.log(data);
    });
  });

  req.on('error', (error) => {
    console.error(error);
  });

  req.write(JSON.stringify(postData));
  req.end();
}

File: k6/pre.py
import http.client
import json
from random import randint
from datetime import datetime, timedelta
from time import sleep

hostname = "localhost"
port = 8080
path = "/api/v1/ad"
headers = {"Content-Type": "application/json"}


def random_int(min, max):
    return randint(min, max)


gender_sets = [[], ["M"], ["F"]]
country_sets = [[], ["TW"], ["JP"], ["TW", "JP"]]
platform_sets = [
    [],
    ["android"],
    ["ios"],
    ["web"],
    ["android", "ios"],
    ["ios", "web"],
    ["web", "android"],
    ["android", "ios", "web"],
]

connection = http.client.HTTPConnection(hostname, port)

for i in range(1000):
    age_start = random_int(1, 100)
    now = datetime.now()
    today_start = datetime(now.year, now.month, now.day)
    today_end = datetime(now.year, now.month, now.day, 23, 59, 59, 999)
    postData = {
        "title": f"AD {i}",
        "startAt": today_start.isoformat() + 'Z',
        "endAt": today_end.isoformat() + 'Z',
        "conditions": [
            {
                "ageStart": age_start,
                "ageEnd": random_int(age_start, 100),
                "gender": gender_sets[random_int(0, len(gender_sets) - 1)],
                "country": country_sets[random_int(0, len(country_sets) - 1)],
                "platform": platform_sets[random_int(0, len(platform_sets) - 1)],
            }
        ],
    }

    connection.request("POST", path, json.dumps(postData), headers)
    response = connection.getresponse()
    data = response.read().decode()
    print(data)
    sleep(0.1)

connection.close()
